grafana-plugin/src/pages/schedules/Schedules.tsx

```diff
@@ -394,18 +404,37 @@ class SchedulesPage extends React.Component<SchedulesPageProps, SchedulesPageSta
     };
   };
 
+  getFilteredSchedules = () => {
+    const { scheduleStore } = this.props.store;
+    const { filters } = this.state;
+
+    return scheduleStore
+      .getSearchResult()
+      .filter(
+        (schedule) =>
+          filters.status === 'all' ||
+          (filters.status === 'used' && schedule.number_of_escalation_chains) ||
+          (filters.status === 'unused' && !schedule.number_of_escalation_chains)
+      );
+  };
+
   handleSchedulesFiltersChange = (filters: SchedulesFiltersType) => {
-    this.setState({ filters }, this.debouncedUpdateSchedules);
+    this.setState({ filters }, () => this.debouncedUpdateSchedules(filters));
   };
 
-  applyFilters = () => {
-    const { filters } = this.state;
-    const { store } = this.props;
-    const { scheduleStore } = store;
-    scheduleStore.updateItems(filters);
+  applyFilters = (filters: SchedulesFiltersType) => {
+    const { scheduleStore } = this.props.store;
+
+    scheduleStore.updateItems(filters).finally(() => {
```

Comment:
I think it's better to use  `then` because `finally` works for both success and failed requests